internal/pkg/task/config_runner_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package task
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/copilot-cli/internal/pkg/aws/ecs"
+	"github.com/aws/copilot-cli/internal/pkg/task/mocks"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestNetworkConfigRunner_Run(t *testing.T) {
+	testCases := map[string]struct {
+		count     int
+		groupName string
+
+		subnets        []string
+		securityGroups []string
+
+		mockClusterGetter func(m *mocks.MockdefaultClusterGetter)
+		mockStarter       func(m *mocks.MocktaskRunner)
+
+		wantedError error
+		wantedARNs  []string
+	}{
+		"failed to get clusters": {
+			mockClusterGetter: func(m *mocks.MockdefaultClusterGetter) {
+				m.EXPECT().DefaultCluster().Return("", errors.New("error getting default cluster"))
+			},
+			mockStarter: func(m *mocks.MocktaskRunner) {
+				m.EXPECT().RunTask(gomock.Any()).Times(0)
+			},
+			wantedError: fmt.Errorf("get default cluster: error getting default cluster"),
+		},
+		"failed to kick off task": {
+			count:     1,
+			groupName: "my-task",
+
+			subnets:        []string{"subnet-1", "subnet-2"},
+			securityGroups: []string{"sg-1", "sg-2"},
+
+			mockClusterGetter: func(m *mocks.MockdefaultClusterGetter) {
+				m.EXPECT().DefaultCluster().Return("cluster-1", nil)
+			},
+			mockStarter: func(m *mocks.MocktaskRunner) {
+				m.EXPECT().RunTask(ecs.RunTaskInput{
```

Comment:
nit: we can use `gomock.Any()` here for the failure scenario since the test case tests the behavior of what happens on any run task err.

Reply:
! will address this in the next PR on task package